env.yml

```diff
@@ -3,7 +3,7 @@ channels:
   - conda-forge
   - defaults
 dependencies:
-  - python
+  - python=3.10
```

Comment:
Thanks for the comment. Choosing 3.10 explicitly was originally done on purpose, to avoid adding too much variability in the package resolving and making it easier to debug potential conflicts. (What you see is what you get, kind of.) I have seen many people actually name their conda environments with the python version on which it is built, especially if you have multiple environments on the same machine.
I have done a couple of dry-run checks to see how the different options compare in my case,
- `python` gives 3.11.0 and does not work with the current set of requirements specified in `env.yml`, but I guess will be resolved in the not too far future
- `python=3.10` gives 3.10.8
- `python<=3.10` gives 3.9.15
- `python<3.11` gives 3.10.8

So I guess from these, the closest one to the suggestion would be `<3.11`, and users who'd like to use a very specific version older than 3.10 need to specify that, as `<3.11` defaults to recent 3.10, which offers fantastic error messages [and other nice improvements over 3.9.](https://docs.python.org/3/whatsnew/3.10.html).

In any case, even if 3.8,3.9,3.10 are all compatible with the current version, what will be installed when choosing the `env.yml` is only one of them, just like previously, `python` only used to install for example 3.9 and not 3.8 and 3.9.

For packages which are not python, I agree to have ranges as a default, but for python as the fundamental building block of the environment that starts everything, in my opinion we need to make clear which version we are using (= to avoid running into similar problems as before when not being specific enough).

Doing this does not exclude that if someone would like to use something else, he/she can modify this line individually. Here we'd have something that is definitely known to work, and from a pedagogical point of view, it's transparent for new students which version is used when they start from scratch.

Reply:
Thanks for the reply, I see the point try to fix with a certain version in yaml file. My original idea is to show it's compatible with other version, but in this test as you show here would introduce some complexity.

I would merge it. thanks!